L/libjulia/common.jl

```diff
@@ -167,20 +165,31 @@ function configure(version)
 
     # These are the platforms we will build for by default, unless further
     # platforms are passed in on the command line
-    #
+    platforms = supported_platforms()
+    # For now skip FreeBSD...
+    filter!(!Sys.isfreebsd, platforms)
+    if version < v"1.5"
+        # in Julia <= 1.4 skip all musl builds
+        filter!(p -> !(Sys.islinux(p) && libc(p) == "musl"), platforms)
+        # in Julia <= 1.4 skip 32bit ARM builds
+        filter!(p -> !(Sys.islinux(p) && arch(p) == "armv7l"), platforms)
```

Comment:
This is a bit unfortunate, but I got this error: when building for ARM 32bit
```
...
[00:34:06]  arm-linux-gnueabihf-g++ -shared -pipe -fPIC -fno-rtti -pedantic -D_FILE_OFFSET_BITS=64 -I/workspace/destdir/include  -O3 -ggdb2 -falign-functions -D_GNU_SOURCE -I. -I/workspace/srcdir/julia-1.4.2/src -I/workspace/srcdir/julia-1.4.2/src/flisp -I/workspace/srcdir/julia-1.4.2/src/support -I/workspace/destdir/include -I/workspace/srcdir/julia-1.4.2/usr/include -DLIBRARY_EXPORTS -I/workspace/srcdir/julia-1.4.2/deps/valgrind -Wall -Wno-strict-aliasing -fno-omit-frame-pointer -fvisibility=hidden -fno-common -Wpointer-arith -Wundef -Wno-unused-result -DJL_BUILD_ARCH='"arm"' -DJL_BUILD_UNAME='"Linux"' -I -DLLVM_SHLIB "-DJL_SYSTEM_IMAGE_PATH=\"../lib/julia/sys.so\"" ./jloptions.o ./runtime_ccall.o ./rtutils.o ./codegen.o ./llvm-ptls.o ./jltypes.o ./gf.o ./typemap.o ./ast.o ./builtins.o ./module.o ./interpreter.o ./symbol.o ./dlload.o ./sys.o ./init.o ./task.o ./array.o ./dump.o ./staticdata.o ./toplevel.o ./jl_uv.o ./datatype.o ./simplevector.o ./runtime_intrinsics.o ./precompile.o ./threading.o ./partr.o ./stackwalk.o ./gc.o ./gc-debug.o ./gc-pages.o ./gc-stacks.o ./method.o ./jlapi.o ./signal-handling.o ./safepoint.o ./timing.o ./subtype.o ./crc32c.o ./APInt-C.o ./processor.o ./jitlayers.o ./debuginfo.o ./disasm.o ./llvm-simdloop.o ./llvm-muladd.o ./llvm-final-gc-lowering.o ./llvm-pass-helpers.o ./llvm-late-gc-lowering.o ./llvm-lower-handlers.o ./llvm-gc-invariant-verifier.o ./llvm-propagate-addrspaces.o ./llvm-multiversioning.o ./llvm-alloc-opt.o ./cgmemmgr.o ./llvm-api.o -Wl,-rpath,'$ORIGIN/julia' -Wl,-rpath,'$ORIGIN' -Wl,-z,origin -o /workspace/srcdir/julia-1.4.2/usr/lib/libjulia.so.1.4 -Wl,-Bdynamic -Wl,-no-undefined -L/workspace/destdir/lib -Wl,-Bsymbolic-functions -Wl,--whole-archive ./flisp/libflisp.a -Wl,--whole-archive ./support/libsupport.a -L/workspace/srcdir/julia-1.4.2/usr/lib -L/workspace/srcdir/julia-1.4.2/usr/lib /workspace/destdir/lib/libuv.a -lutf8proc -Wl,--no-whole-archive -L/workspace/destdir/lib -lLLVM-8jl -Wl,--no-as-needed -ldl -lrt -lpthread -Wl,--export-dynamic,--as-needed,--no-whole-archive -Wl,--version-script=/workspace/srcdir/julia-1.4.2/src/julia.expmap /workspace/srcdir/julia-1.4.2/usr/lib/libunwind-generic.a /workspace/srcdir/julia-1.4.2/usr/lib/libunwind.a -Wl,-soname=libjulia.so.1
[00:34:07] ./gc.o: In function `combine_thread_gc_counts':
[00:34:07] /workspace/srcdir/julia-1.4.2/src/gc.c:998: undefined reference to `__atomic_load_8'
[00:34:07] /workspace/srcdir/julia-1.4.2/src/gc.c:999: undefined reference to `__atomic_load_8'
[00:34:07] /workspace/srcdir/julia-1.4.2/src/gc.c:1000: undefined reference to `__atomic_load_8'
[00:34:07] /workspace/srcdir/julia-1.4.2/src/gc.c:1001: undefined reference to `__atomic_load_8'
[00:34:07] /workspace/srcdir/julia-1.4.2/src/gc.c:1002: undefined reference to `__atomic_load_8'
[00:34:07] ./gc.o:/workspace/srcdir/julia-1.4.2/src/gc.c:1003: more undefined references to `__atomic_load_8' follow
[00:34:07] ./gc.o: In function `jl_gc_collect':
[00:34:07] /workspace/srcdir/julia-1.4.2/src/gc.c:3076: undefined reference to `__atomic_fetch_add_8'
[00:34:07] collect2: error: ld returned 1 exit status
[00:34:07] make[1]: *** [Makefile:331: /workspace/srcdir/julia-1.4.2/usr/lib/libjulia.so.1.4] Error 1
```
Does anybody know if Julia 1.4.2 was supposed to work on ARM32bit?

Reply:
You might need to explicitly link against `-latomic`?

Reply:
Tried it, couldn't get it to work, then you told me not to bother ... ;-)

Reply:
Ah right! ;)